django/contrib/admin/checks.py

```diff
@@ -28,6 +29,22 @@ def _issubclass(cls, classinfo):
         return False
 
 
+def _contains_subclass(subclass_path, candidate_paths):
+    """
+    Check if dotted subclass module path is found among a list of candidate
+    paths.
+    """
+    subclass = import_string(subclass_path)
+    for path in candidate_paths:
+        try:
+            candidate_cls = import_string(path)
+        except ImportError:
```

Comment:
Why should an import error pass silently? This doesn't seem to be covered by a test. Assuming it's not needed, this could be simplified to `return any(_issubclass(import_string(path), subclass) for path in candidate_paths)`.

Reply:
> Why should an import error pass silently?

@timgraham the idea was that we probably want the `ImportError` to be raised from the `MIDDLEWARE` loading code then from an admin check. I agree that it should be tested though.

Reply:
Agreed. How about something like?

```suggestion
@override_settings(MIDDLEWARE=[
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.contrib.does.not.Exist',
])  
def test_admin_check_does_not_raise_import_error(self):
    self.assertEqual(admin.checks.check_dependencies(), []) 
```

Reply:
Need to put the nonexistent classes before the valid ones, otherwise the code won't get to "does.not.exist".